kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Request.scala

```diff
@@ -0,0 +1,24 @@
+package hydra.kafka.model
+
+import java.time.Instant
+
+import hydra.core.marshallers.StreamType
+import org.apache.avro.Schema
+
+sealed trait ContactMethod
+
+final case class Email(address: String) extends ContactMethod
+final case class Slack(channel: String) extends ContactMethod
+
+final case class Schemas(key: Schema, value: Schema)
+
+case class TopicMetadataV2Request(
+                                  subject: String,
+                                  schemas: Schemas,
+                                  streamType: StreamType,
+                                  deprecated: Boolean,
+                                  dataClassification: String,
+                                  contact: List[ContactMethod],
+                                  createdDate: Instant,
+                                  parentSubjects: List[String],
+                                  notes: Option[String])
```

Comment:
I am thinking that we may be benefit from a few stronger types here. I am thinking `contact` can be `NonEmptyList`, `dataClassification` can be an ADT (I believe), and subject can be a concrete `Subject` or `HydraSubject` or something. Then we'd have `subject: Subject` and `parentSubjects: List[Subject]`. The benefit I see here is mainly encapsulating all of the logic for validation within the `Subject` class itself rather than needing to live in a separate validator. This is kind of related with what I was saying about "parse not validate." If I see a `String` on a field that I know would need to be validated, I don't have a way of easily being able to tell if it has been validated yet or not. Something like `Subject` on the other hand has all of the validation needed within the constructor of the type so I can be confident that it is good to go.

Long comment for what I think will be a pretty minor change haha. We can chat more, but just wanted to jot this idea down while I was thinking about it.

Reply:
Agreed and fixed.